keras/optimizers.py

```diff
@@ -69,6 +70,9 @@ def get_updates(self, params, constraints, loss):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if hasattr(self, 'normalize_gradients') and self.normalize_gradients:
+            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads])) + K.epsilon()
```

Comment:
Sure. I was curious to experiment with some of those changes myself. Might as well test it on simple example.